plugins/scaffolder/src/components/fields/EntityPicker/EntityPicker.tsx

```diff
@@ -109,3 +107,70 @@ export const EntityPicker = (props: EntityPickerProps) => {
     </FormControl>
   );
 };
+
+/**
+ * Converts a boolean value `true` to `CATALOG_FILTER_EXISTS`.
+ * Returns the input value if it is not a boolean `true`.
+ *
+ * @param value - The value to convert.
+ * @returns The converted value.
+ */
+function convertTrueToExists(value: string | boolean): string | symbol {
+  if (value === true) {
+    return CATALOG_FILTER_EXISTS;
+  }
+  return value?.toString();
+}
+
+/**
+ * Converts schema filters to entity filter query, replacing boolean `true` values
+ * with the constant `CATALOG_FILTER_EXISTS`.
+ *
+ * @param schemaFilters - An object containing schema filters with keys as filter names
+ * and values as filter values.
+ * @returns An object with the same keys as the input object, but with `true` boolean values
+ * transformed to `CATALOG_FILTER_EXISTS` symbol.
+ */
+function convertSchemaFiltersToQuery(
+  schemaFilters: Record<string, string | boolean | (string | boolean)[]>,
+): Record<string, string | symbol | (string | symbol)[]> {
+  const query: EntityFilterQuery = {};
+
+  for (const [key, value] of Object.entries(schemaFilters)) {
+    if (Array.isArray(value)) {
+      query[key] = value.map(convertTrueToExists) as string[];
```

Comment:
Not really supposed to be converting the array values right?

Reply:
good catch thanks